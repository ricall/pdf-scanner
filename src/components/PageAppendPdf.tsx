import React from 'react';
import { Button, Paper } from "@material-ui/core";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";

const useStyles = makeStyles((theme: Theme) => createStyles({
    paper: {
        width: 270,
        height: 380,
        overflow: 'hidden',
        border: '3px dashed rgba(0, 0, 0, 0.80)',
    },
    upload: {
        width: '100%',
        height: '100%',
    },
}));

export interface PageDropProps {
    label: string,
    onAddDocument: (name: string, document: ArrayBuffer) => void,
}

export const PageAppendPdf: React.FC<PageDropProps> = ({ label, onAddDocument }) => {
    const classes = useStyles();
    const onFileChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
        const fileList = event.target.files;
        if (!fileList) {
            return;
        }
        const file = fileList[0]
        onAddDocument(file.name, await file.arrayBuffer());
    };
    return (
        <Paper className={classes.paper} elevation={0}>
            <input accept="application/pdf"
                   style={{ display: 'none ' }}
                   id="file-upload"
                   type="file"
                   onChange={onFileChange}/>

            <label htmlFor="file-upload">
                <Button className={classes.upload} component="span">
                    {label}
                </Button>
            </label>
        </Paper>
    );
}