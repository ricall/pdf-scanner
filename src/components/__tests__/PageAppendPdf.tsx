import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { PageAppendPdf } from "../PageAppendPdf";

describe('PageAppendPdf component', () => {

    it('can be rendered', () => {
        const wrapper = shallow(<PageAppendPdf label="label" onAddDocument={jest.fn()} />);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

});
