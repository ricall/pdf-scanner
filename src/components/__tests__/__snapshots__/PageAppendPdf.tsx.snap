// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageAppendPdf component can be rendered 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-paper-1"
  elevation={0}
>
  <input
    accept="application/pdf"
    id="file-upload"
    onChange={[Function]}
    style={
      Object {
        "display": "none ",
      }
    }
    type="file"
  />
  <label
    htmlFor="file-upload"
  >
    <WithStyles(ForwardRef(Button))
      className="makeStyles-upload-2"
      component="span"
    >
      label
    </WithStyles(ForwardRef(Button))>
  </label>
</WithStyles(ForwardRef(Paper))>
`;
